openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -217,16 +234,286 @@ def _add_serializer_errors(update, serializer, field_errors):
     """
     if not serializer.is_valid():
         errors = serializer.errors
-        for key, value in errors.iteritems():
-            if isinstance(value, list) and len(value) > 0:
-                developer_message = value[0]
-            else:
-                developer_message = "Invalid value: {field_value}'".format(field_value=update[key])
+        for key, error in errors.iteritems():
+            field_value=update[key]
             field_errors[key] = {
-                "developer_message": developer_message,
-                "user_message": _("Value '{field_value}' is not valid for field '{field_name}'.".format(
-                    field_value=update[key], field_name=key)
-                )
+                "developer_message": "Value '{field_value}' is not valid for field '{field_name}': {error}".format(
```

Comment:
Will this raise a unicode error if field value or the error message contains non-ASCII unicode?